torch/fx/experimental/symbolic_shapes.py

```diff
@@ -2654,8 +2647,10 @@ def is_dim(src):
         # pruned).  In this case, only Dynamo knows about these arguments.
         def track_symint(source, val, constraint=None):
             if isinstance(val, SymInt):
-                s = val.node.expr
+                val = val.node.maybe_as_int() or val
```

Comment:
This will wrongly put 0 into val

Reply:
Not sure I get it. The only problem I can see is if `val.node.maybe_as_int()` returns 0, e.g. if there's a guard `x == 0` (is this possible?).

The way I see it, there are 2 possibilities:
1. `val.node.maybe_as_int()` returns `None`, then `val` will stay as is
2. `val.node.maybe_as_int()` returns an int (other than 0), then `val` will be an int

Am I missing something?